baseplate/sidecars/event_publisher.py

```diff
@@ -161,6 +165,33 @@ def publish(self, payload: SerializedBatch) -> None:
         raise MaxRetriesError("could not sent batch")
 
 
+def build_batch_and_publish(
+    event_queue: MessageQueue, batcher: TimeLimitedBatch, publisher: BatchPublisher, timeout: float
+) -> bytes:
+    # Helper that continuously polls for messages, then batches and publishes them
+    while True:
+        message: Optional[bytes]
+
+        try:
+            message = event_queue.get(timeout)
+        except TimedOutError:
+            message = None
+
+        try:
+            batcher.add(message)
+            continue
```

Comment:
I see what you are doing here. The implementation of `TimeLimitedBatch` forces you to do it this way. But is smells wrong. It would be better if the `TimeLimitedBatch` class had an `is_complete()` function to tell you if its age is over its max_age, then you could use a much neater if statement.

Reply:
this is great advice! I want to avoid making too many unnecessary changes that are out of scope of the purpose of this change, but I think that's a good idea of something to improve in the future